cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -375,6 +376,35 @@ std::array<std::array<std::vector<char>, NUM_SYMBOL_GROUPS>, TT_NUM_STATES> cons
 
 // The DFA's starting state
 constexpr auto start_state = static_cast<StateT>(TT_OOS);
+
+template <typename SymbolT>
+auto get_sgid_lut(SymbolT delim)
+{
+  symbol_groups[static_cast<std::uint8_t>(dfa_symbol_group_id::DELIMITER_CHAR)] = {delim};
+  return symbol_groups;
+}
+
+auto get_transition_table(stack_behavior_t stack_behavior)
+{
+  // Transition table specialized on the choice of whether to reset on newlines
+  return (stack_behavior == stack_behavior_t::ResetOnDelimiter)
+           ? to_stack_op::resetting_transition_table
+           : to_stack_op::transition_table;
+}
+
+template <typename SymbolT>
+auto get_translation_table(SymbolT delim, stack_behavior_t stack_behavior)
+{
+  // Translation table specialized on the choice of whether to reset on newlines
+  if (stack_behavior == stack_behavior_t::ResetOnDelimiter) {
+    resetting_translation_table[0][6] = {delim};
+    resetting_translation_table[1][6] = {delim};
+    resetting_translation_table[2][6] = {delim};
```

Comment:
It feels quite fragile to use hardcoded numerical values of the table and modify them here. I'd suggest we either use the enum values or, likely even better, construct the whole table in this function rather than modifying it.

Reply:
Fixed! The translation table is initialized inside the function and returned.